src/libraries/System.Private.CoreLib/src/System/Globalization/Ordinal.cs

```diff
@@ -236,7 +238,51 @@ internal static int IndexOfOrdinalIgnoreCase(ReadOnlySpan<char> source, ReadOnly
                 return CompareInfo.NlsIndexOfOrdinalCore(source, value, ignoreCase: true, fromBeginning: true);
             }
 
-            return OrdinalCasing.IndexOf(source, value);
+            // if value starts with an ASCII char we can use a vectorized path
+            ref char valueRef = ref MemoryMarshal.GetReference(value);
+            char valueChar = valueRef;
+
+            if (!char.IsAscii(valueChar))
+            {
+                // Fallback to a more non-ASCII friendly version
+                return OrdinalCasing.IndexOf(source, value);
+            }
+
+            ref char searchSpace = ref MemoryMarshal.GetReference(source);
+            int valueLength = value.Length;
+            int searchSpaceLength = source.Length;
+
+            do
+            {
+                // if val is either [a..z] or [A..Z] - search for its lower and upper counter parts using IndexOfAny
+                // otherwise use just plain IndexOf
+                int candidatePos = (uint)((valueChar | 0x20) - 'a') <= 'z' - 'a' ?
+                    SpanHelpers.IndexOfAny(ref searchSpace, (char)(valueChar & ~0x20), (char)(valueChar | 0x20), searchSpaceLength) :
+                    SpanHelpers.IndexOf(ref searchSpace, valueChar, searchSpaceLength);
+
+                if ((candidatePos == -1) || (searchSpaceLength - candidatePos) < valueLength)
```

Comment:
```suggestion
                if ((candidatePos < 0) || (searchSpaceLength - candidatePos) < valueLength)
```
Or does the JIT now do this optimization (`test` <-> `cmp`) now for us?

Reply:
I don't think it's legal for jit, it doesn't know that the only possible negative value is -1, Thanks, but my recent rewriting based on https://github.com/dotnet/runtime/blob/20c4bde6b1b8b7afb173abf730a10eb10c6be55e/src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Char.cs#L41-L66 already contains this fix